pkgs/swift2objc/lib/src/transformer/transformers/transform_property.dart

```diff
@@ -55,24 +55,20 @@ List<String> _generateGetterStatemenets(
   final wrappedInstanceProperty =
       '${wrappedClassInstance.name}.${originalProperty.name}';
 
-  if (originalProperty.type.isObjCRepresentable) {
-    assert(originalProperty.type.id == transformedProperty.type.id);
+  if (originalProperty.type.id == transformedProperty.type.id) {
```

Comment:
Can this check go in `generateWrappedValue` too?

Reply:
well `generateWrappedValue` is supposed to wrap the value, so it's not its responsibility to check if it should wrap the value or not. I think it makes more sense to perform this check before calling it. Same for `generateUnwrappedValue`. Wdyt?

Reply:
The goal is to reduce code duplication. It seems like the sort of check that every call site is going to need to do, so if you can cleanly move the check inside then you should. You can always rename the function to indicate it might not need to wrap the value, eg `maybeWrapValue`.

Reply:
Sounds good. Done. It's not quite the same check but it achieves the same thing.